lib/src/transformer/transformer.dart

```diff
@@ -68,8 +81,9 @@ class _TransformAssetWriter implements AssetWriter {
   _TransformAssetWriter(this.transform);
 
   @override
-  Future writeAsString(build.Asset asset, {Encoding encoding: UTF8}) async =>
-      transform.addOutput(_toBarbackAsset(asset));
+  Future writeAsString(build.Asset asset, {Encoding encoding: UTF8}) async {
```

Comment:
It does have the interesting side-effect of running `transform.addOutput` asynchronously as well, which is a bit interesting.